packages/batch/tests/unit/BatchProcessor.test.ts

```diff
@@ -418,4 +426,42 @@ describe('Class: BatchProcessor', () => {
       );
     });
   });
+
+  describe('Batch processing with Lambda context', () => {
+    test('Batch processing when context is provided and handler accepts', async () => {
+      // Prepare
+      const firstRecord = sqsRecordFactory('success');
+      const secondRecord = sqsRecordFactory('success');
+      const records = [firstRecord, secondRecord];
+      const processor = new BatchProcessor(EventType.SQS);
+
+      // Act
+      processor.register(records, handlerWithContext, options);
+      const processedMessages = await processor.process();
+
+      // Assess
+      expect(processedMessages).toStrictEqual([
+        ['success', firstRecord.body, firstRecord],
+        ['success', secondRecord.body, secondRecord],
+      ]);
+    });
+
+    test('Batch processing when context is provided and handler does not accept', async () => {
+      // Prepare
+      const firstRecord = sqsRecordFactory('success');
+      const secondRecord = sqsRecordFactory('success');
+      const records = [firstRecord, secondRecord];
+      const processor = new BatchProcessor(EventType.SQS);
+
+      // Act
+      processor.register(records, sqsRecordHandler, options);
```

Comment:
Should we call `handlerwithContext` here? Is the test designed to pass options even though we don't expect it to be used?

Reply:
This test is to check that if users register with context, even if the handler doesn't accept context it will still run. I think TS/JS is built so that the handler (without options parameter defined) will still work even if the processor tries to pass options into it, but I wasn't aware of this so I checked the behavior. This resolves the last conversation as well, so I'll make the change to simplify the logic, but I think the test here is fine?

Reply:
Yep, makes sense, we can keep it.